org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/apt/dispatch/BatchFilerImpl.java

```diff
@@ -43,13 +48,32 @@ public class BatchFilerImpl implements Filer {
 	protected final BatchProcessingEnvImpl _env;
 	protected final JavaFileManager _fileManager;
 	protected final HashSet<URI> _createdFiles;
+	protected String _moduleName;
+	protected String _encoding;
 
-	public BatchFilerImpl(BaseAnnotationProcessorManager dispatchManager, BatchProcessingEnvImpl env)
+	public BatchFilerImpl(BaseAnnotationProcessorManager dispatchManager, BatchProcessingEnvImpl env, Main main)
 	{
 		this._dispatchManager = dispatchManager;
 		this._fileManager = env._fileManager;
 		this._env = env;
 		this._createdFiles = new HashSet<>();
+		this._encoding = main.getDefaultEncoding();
+		if (this._fileManager.hasLocation(StandardLocation.SOURCE_PATH)) {
+			try {
+				for (JavaFileObject javaFileObject : this._fileManager.list(StandardLocation.SOURCE_PATH, "", //$NON-NLS-1$
+						Collections.singleton(Kind.SOURCE), false)) {
+					if (javaFileObject.getName().equals(IModule.MODULE_INFO_JAVA)) {
```

Comment:
small question about this... I was just checking the JLS for this for Java 9... and they seem to suggest `module-info.jav` is also a potentially valid name?

> If and only if packages are stored in a file system ([§7.2](https://docs.oracle.com/javase/specs/jls/se9/html/jls-7.html#jls-7.2)), the host system may choose to enforce the restriction that it is a compile-time error if a module declaration is not found in a file under a name composed of module-info plus an extension (such as .java or .jav).

This seems a bit odd to me, do we know if we're supposed to respect that as well (both on your side and my side)?